.github/workflows/codacy-analysis.yml

```
@@ -0,0 +1,46 @@
+# This workflow checks out code, performs a Codacy security scan
+# and integrates the results with the
+# GitHub Advanced Security code scanning feature.  For more information on
+# the Codacy security scan action usage and parameters, see
+# https://github.com/codacy/codacy-analysis-cli-action.
+# For more information on Codacy Analysis CLI in general, see
+# https://github.com/codacy/codacy-analysis-cli.
+
+name: Codacy Security Scan
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  codacy-security-scan:
+    name: Codacy Security Scan
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the repository to the GitHub Actions runner
+      - name: Checkout code
+        uses: actions/checkout@v2
```

They now recommend using the latest version of everything (see example [here](https://github.com/codacy/codacy-analysis-cli-action#integration-with-github-code-scanning)):
```suggestion
        uses: actions/checkout@main
```

---

Actually it seems GitHub Actions does [not download the submodules](https://github.com/actions/checkout#usage) by default, which is probably what is causing that error:
```suggestion
        uses: actions/checkout@main
        with:
          submodules: recursive
```

---

```suggestion
        uses: actions/checkout@v2
        with:
          submodules: recursive
```

---

The latest version is actually v3:
```suggestion
        uses: actions/checkout@v3
```